### ISSUE TITLE
Need support to enforce HTTP/1.1 API call in http component in benthos

### ISSUE THREAD

**Issue by jeetvachhani285** [redpanda-data/benthos#117]
While I am calling API using **http** component i am getting `stream error: stream ID 6107; HTTP_1_1_REQUIRED; received from peer` error in benthos.

It will be working fine if there is an intermediary, such as a proxy, firewall, or security software (e.g., Netskope). It might be possible because it will be enforcing or preferring HTTP/1.1 over HTTP/2  but without that it throws error.

Config file: 

```
http:
    url: https://<cloud-instance>/BeyondTrust/api/public/v3/Auth/SignAppIn
    verb: POST
    rate_limit: api_ratelimit
    timeout: 45s
    extract_headers:
      include_prefixes:
        - set-cookie
```
Here you can see the error which i am getting.

```
level=error msg="Failed to call SignIn API due to https://<cloud-instance>/BeyondTrust/api/public/v3/Auth/SignAppIn: Post \"https://<cloud-instance>/BeyondTrust/api/public/v3/Auth/SignAppIn\": Post \"https://<cloud-instance>/BeyondTrust/api/public/v3/Auth/Connect/Token\": stream error: stream ID 521347; HTTP_1_1_REQUIRED; received from peer" @service=benthos label="" path=root.input.broker.inputs.0.processors.3.switch.processors.2.catch.0
``` 

So is there any way to by which we can enforce HTTP/1.1 for API call  may be by adding some an advanced flag for this in benthos?

**Comment by mihaitodor** [comment]
Hey @jeetvachhani285, thank you for raising this issue! I'm not sure how to reproduce it on my end, but I believe we can do what you're asking for by exposing a flag to set Transport.TLSNextProto to an empty non-nil map for HTTP clients as described here: https://pkg.go.dev/net/http?utm_source=godoc#hdr-HTTP_2. Please let us know if that makes sense and if you can help us test it.

**Comment by jeetvachhani285** [comment]
> Hey [@jeetvachhani285](https://github.com/jeetvachhani285), thank you for raising this issue! I'm not sure how to reproduce it on my end, but I believe we can do what you're asking for by exposing a flag to set Transport.TLSNextProto to an empty non-nil map for HTTP clients as described here: https://pkg.go.dev/net/http?utm_source=godoc#hdr-HTTP_2. Please let us know if that makes sense and if you can help us test it.
> 
> PS: Would you mind checking if setting the `GODEBUG=http2client=0` environment variable helps? Also, could you please check if you can use `curl` to execute a POST request to that endpoint and share with us what flags you used?

@mihaitodor I have checked the [doc](https://pkg.go.dev/net/http#hdr-HTTP_2) you have shared. Ideally it should work but while i am checking after setting env variable GODEBUG=http2client=0 I am getting below error may be server requires TLS negotiation.

```
Failed to call SignIn API due to https://<cloud-instance>/BeyondTrust/api/public/v3/Auth/SignAppIn: Post "https://<cloud-instance>/BeyondTrust/api/public/v3/Auth/SignAppIn": Post "https://<cloud-instance>/BeyondTrust/api/public/v3/Auth/Connect/Token": local error: tls: no renegotiation
```

**Comment by mihaitodor** [comment]
Hey @jeetvachhani285, would you mind testing the following (leave `GODEBUG=http2client=0` set):

```yaml
http:
  ...
  tls:
    enabled: true
    enable_renegotiation: true
```

If that doesn't work, please also try this:

```yaml
http:
  ...
  tls:
    enabled: true
    enable_renegotiation: true
    skip_cert_verify: true
```

(replace `...` with your current `http` fields)

If you have a self-signed cert and it doesn't work without `skip_cert_verify: true`, then, for security reasons, you'll probably want to set `skip_cert_verify: false` and configure [`tls.root_cas`](https://docs.redpanda.com/redpanda-connect/components/processors/http/#tls-root_cas) (or [`tls.root_cas_file`](https://docs.redpanda.com/redpanda-connect/components/processors/http/#tls-root_cas_file)).

**Comment by jeetvachhani285** [comment]
> Hey [@jeetvachhani285](https://github.com/jeetvachhani285), would you mind testing the following (leave `GODEBUG=http2client=0` set):
> 
> http:
>   ...
>   tls:
>     enabled: true
>     enable_renegotiation: true
> If that doesn't work, please also try this:
> 
> http:
>   ...
>   tls:
>     enabled: true
>     enable_renegotiation: true
>     skip_cert_verify: true
> (replace `...` with your current `http` fields)
> 
> If you have a self-signed cert and it doesn't work without `skip_cert_verify: true`, then, for security reasons, you'll probably want to set `skip_cert_verify: false` and configure [`tls.root_cas`](https://docs.redpanda.com/redpanda-connect/components/processors/http/#tls-root_cas) (or [`tls.root_cas_file`](https://docs.redpanda.com/redpanda-connect/components/processors/http/#tls-root_cas_file)).

Thanks @mihaitodor , I will check and let you know.

**Comment by jeetvachhani285** [comment]
@mihaitodor Keeping env `GODEBUG=http2client=0` and adding the tls settings  is working fine without proxy as well

```
http:
  ...
  tls:
    enabled: true
    enable_renegotiation: true
```

**Comment by mihaitodor** [comment]
Sounds good, thank you for confirming!

**Comment by jeetvachhani285** [comment]
Hey @mihaitodor This things are working locally but to work globally as you mentioned in first comment we require to expose a flag to set Transport.TLSNextProto to an empty non-nil map for HTTP clients and for that you require changes in redpanda. Am i correct ?

**Comment by mihaitodor** [comment]
Yes, that's right. I added an item to the backlog to expose that as a config flag.
